app.territoiresentransitions.fr/src/components/tracking/Matomo.svelte

```diff
@@ -0,0 +1,86 @@
+<script context="module">
+    /*
+        Expose an object to handle Matomo asynchronously.
+
+        This component exposes an interface with existing methods on Matomo Tracker
+        (cf. https://developer.matomo.org/api-reference/tracking-javascript). Instead of
+        calling a method immediately, it stores it in a queue.
+     */
+    import { writable } from 'svelte/store'
+
+    // We only implement methods from Matomo API that we are using on this application.
+    // You can check the documentation to add more Tracking API methods:
+    // https://developer.matomo.org/api-reference/tracking-javascript.
+    const methods = [
+        'trackPageView'
+    ]
+
+    const queuedEvents = writable([])
+
+    export const asyncMatomo = methods
+        // ['trackPageView', (...args) => ]
+        .map(method =>
+            ([method, (...args) => queuedEvents.update(
+                alreadyQueued => [...alreadyQueued, [method, args]])]
+            )
+        )
+        .reduce((acc, [key, value]) => ({...acc, [key]: value}), {})

```

Comment:
Je crois que je préfère les trucs plus explicites que le reduce notamment pour que asyncMatomo soit typé, mais vu que l'on s'en sert que à un endroit c'est bon pour moi. On peut en discuter aussi
```
type MatomoMethod = (...args: any[]) => void;

const asyncMatomo = new Map<string, MatomoMethod>()

for (let name of methodNames) {
    let method: MatomoMethod = ...
    asyncMatomo.set(name, method)
}
````

Reply:
@derfurth Ah oui, je vois ce que tu veux dire. Je propose de merger cette PR, d'en discuter et d'itérer sur une deuxième PR (vu que je dois repasser dessus)